### PR TITLE
commiting changes 2-15

### PR DESCRIPTION
**Pull request by benphillips989**
Hi,

I did a bunch of work on the database code this weekend and would like to merge to dev.
Below is a summary of what I did so far.

Thanks,
Ben

added method to Table_Init.py that checks if a table already exists,
this is case sensitive

updated Table_Init so that it uses the stub methods described in
## added unit test to verify that check_table_exists works correctly

began writing DataManager class
    contains a consumer thread
    contains a method to manage the producer side of notifying
        consumer thread
    contains init to create dataQueue and condition object.
        this also passes references of the queue and condition
        object into the consumer thread

```
    this also initializes the database
```

---
## TODO: write tests for DataManager Class

began writing dataQueue class
    basically creates a queue object and exposes methods we would
        like to use that can also be used for validation in the
            future

```
created insert_into_data_queue method
created get_next_item method
created check_empty method
```

---
## created simple test for dataQueue will add some more if needed

added stub methods to DB_Init for getting the database config name
and the home config path, also created a method to get the value
of the 'HOME' environment variable from the OS

changed DB_Init to refer to these methods instead of hard coded
## strings, will need to tie these methods to the config object

updated Table_Insert to use stub methods in DB_Init to find
connection

### CODE REVIEW

**Comment by mcguiremw** [comment]
:+1:

**Comment by jnels124** [comment]
:100:
